### ISSUE TITLE
Flannel seems to have problems with leader election hook

### ISSUE THREAD

**Issue by mbruzek** [juju-solutions/charm-flannel#29]
While debugging a customer problem we see a repeated error in the flannel log. The problem seems to occur when the leader-settings-changed hook runs.

```
2017-01-27 21:21:05 INFO juju-log Reactive main running for hook leader-settings-changed
2017-01-27 21:21:06 INFO juju.worker.upgrader upgrader.go:178 desired tool version: 2.0.2
2017-01-27 21:21:06 INFO juju-log Invoking reactive handler: reactive/flannel.py:167:ready
2017-01-27 21:21:06 INFO leader-settings-changed Traceback (most recent call last):
2017-01-27 21:21:06 INFO leader-settings-changed   File "/var/lib/juju/agents/unit-flannel-4/charm/hooks/leader-settings-changed", line 19, in <module>
2017-01-27 21:21:06 INFO leader-settings-changed     main()
2017-01-27 21:21:06 INFO leader-settings-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/__init__.py", line 78, in main
2017-01-27 21:21:06 INFO leader-settings-changed     bus.dispatch()
2017-01-27 21:21:06 INFO leader-settings-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 434, in dispatch
2017-01-27 21:21:06 INFO leader-settings-changed     _invoke(other_handlers)
2017-01-27 21:21:06 INFO leader-settings-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 417, in _invoke
2017-01-27 21:21:06 INFO leader-settings-changed     handler.invoke()
2017-01-27 21:21:06 INFO leader-settings-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 291, in invoke
2017-01-27 21:21:06 INFO leader-settings-changed     self._action(*args)
2017-01-27 21:21:06 INFO leader-settings-changed   File "/var/lib/juju/agents/unit-flannel-4/charm/reactive/flannel.py", line 171, in ready
2017-01-27 21:21:06 INFO leader-settings-changed     status_set('active', 'Flannel subnet ' + get_flannel_subnet())
2017-01-27 21:21:06 INFO leader-settings-changed   File "/var/lib/juju/agents/unit-flannel-4/charm/reactive/flannel.py", line 223, in get_flannel_subnet
2017-01-27 21:21:06 INFO leader-settings-changed     with open('/run/flannel/subnet.env') as f:
2017-01-27 21:21:06 INFO leader-settings-changed FileNotFoundError: [Errno 2] No such file or directory: '/run/flannel/subnet.env'
2017-01-27 21:21:06 ERROR juju.worker.uniter.operation runhook.go:107 hook "leader-settings-changed" failed: exit status 1
2017-01-27 21:21:06 INFO juju.worker.uniter resolver.go:100 awaiting error resolution for "leader-settings-changed" hook
2017-01-27 21:21:06 INFO juju.worker.uniter resolver.go:100 awaiting error resolution for "leader-settings-changed" hook
```

We need to look into the charm code for this and fix this problem.

**Comment by Cynerva** [comment]
Thanks for opening this @mbruzek. Fortunately I think this has already been fixed by #26, and that fix is included in our latest stable release of CDK. I double-checked that just now to be sure.

I think we can safely close this so I'mma do that. Thanks again!

**Comment by mbruzek** [comment]
Thanks for keeping me honest @Cynerva
